onnxruntime/core/providers/cpu/tensor/pad.cc

```diff
@@ -438,12 +464,46 @@ static Status PadImpl(OpKernelContext* ctx,
           int64_t prePad = reshaped_pad[inner_axis];
           int64_t postPad = reshaped_pad[inner_axis + new_dims_count];
           if (inner_no_pad_size == 1) {
-            PadInnermostAxis(axisStart - prePad, axisStart + prePad, -1 /* inputDelta */, onnxruntime::narrow<size_t>(prePad));
-            PadInnermostAxis(output, output - 2, -1 /* inputDelta */, onnxruntime::narrow<size_t>(postPad));
+            if (mode == Mode::Reflect) {
+              PadInnermostAxis(axisStart - prePad, axisStart + prePad, -1 /* inputDelta */, onnxruntime::narrow<size_t>(prePad));
+              PadInnermostAxis(output, output - 2, -1 /* inputDelta */, onnxruntime::narrow<size_t>(postPad));
+            } else {
```

Comment:
Nit: a comment like `// mode == Mode::Wrap` would be helpful in the else-line (here and similarly below).